volatility3/framework/symbols/windows/extensions/gui.py

```diff
@@ -2,126 +2,333 @@
+                return desktop.get_session_id()
+
+            return None
+
+        def get_desktop(self) -> Optional["GUIExtensions.tagDESKTOP"]:
+            """
+            Attempts to return the host desktop (tagDESKTOP) for this window
+            """
+            try:
+                return self.head.rpdesk.dereference()
+            except exceptions.InvalidAddressException:
+                vollog.debug(
+                    f"Reading the desktop pointer for window {self.vol.offset:#x} caused a page fault"
+                )
+                return None
+
+        def get_process(self) -> Optional["extensions.EPROCESS"]:
+            """
+            Attempts to return the host process (_EPROCESS) for this window
+            """
+            try:
+                return self.head.pti.ppi.Process.dereference()
+            except exceptions.InvalidAddressException:
+                vollog.debug(
+                    f"Reading the process pointer for window {self.vol.offset:#x} caused a page fault"
+                )
+                return None
+
+        def get_window_procedure(self):
+            """
+            Attempts to return the window procedure for this windows
+            """
+            try:
+                # >= 17134
+                if hasattr(self, "subPointer"):
+                    return self.subPointer.lpfnWndProc
+                else:
+                    return self.lpfnWndProc
+            except exceptions.InvalidAddressException:
+                vollog.debug(
+                    f"Invalid window procedure for window {self.vol.offset:#x}"
+                )
+                return None
+
+    # This is copy/paste from UNICODE_STRING in `symbols/windows/extensions/__init__.py`
+    # The versioning of modules would get very ugly if we let different modules share implementations
+    # across different data structures
+    class LARGE_UNICODE_STRING(objects.StructType):
+        """A class for Windows unicode string structures."""
+
+        def get_string(self) -> interfaces.objects.ObjectInterface:
+            # We explicitly do *not* catch errors here, we allow an exception to be thrown
+            # (otherwise there's no way to determine anything went wrong)
+            # It's up to the user of this method to catch exceptions
+
+            # We manually construct an object rather than casting a dereferenced pointer in case
+            # the buffer length is 0 and the pointer is a NULL pointer
+            return self._context.object(
+                self.vol.type_name.split(constants.BANG)[0] + constants.BANG + "string",
+                layer_name=self.Buffer.vol.native_layer_name,
+                offset=self.Buffer,
+                max_length=self.Length,
+                errors="replace",
+                encoding="utf16",
+            )
 
 
 class_types = {
-    "tagWINDOWSTATION": tagWINDOWSTATION,
-    "tagDESKTOP": tagDESKTOP,
+    "tagWINDOWSTATION": GUIExtensions.tagWINDOWSTATION,
```

Comment:
Might be clearer to make this dictionary part of the versioned class, so that it's easy to see where the requirement comes in (and it doesn't get missed on the plugins that use it).